mesh/mesh.go

```diff
@@ -368,7 +368,8 @@ func (msh *Mesh) ProcessLayer(ctx context.Context, layerID types.LayerID) error
 	logger.Info("processing layer")
 
 	// pass the layer to tortoise for processing
-	newVerified := msh.trtl.HandleIncomingLayer(ctx, layerID)
+	msh.trtl.TallyVotes(ctx, layerID)
```

Comment:
based on the description of https://github.com/spacemeshos/go-spacemesh/issues/3575
i'd assume that we don't need this call in mesh.go anymore. in another words, tortoise should advance verified layer based on data from OnBallot(). mesh just needs to check if a new layer is verified and then apply them.

Reply:
it would be too expensive to check if layer crosses threshold after every ballot is counted, there are two points when we need to know when threshold is crossed:
- when node wants to apply layers
- for casting votes